pinot-common/src/main/java/org/apache/pinot/common/utils/LogicalTableUtils.java

```diff
@@ -73,8 +74,11 @@ public static ZNRecord toZNRecord(LogicalTableConfig logicalTableConfig)
     return record;
   }
 
-  public static void validateLogicalTableName(LogicalTableConfig logicalTableConfig, List<String> allPhysicalTables,
-      Set<String> allBrokerTenantNames) {
+  public static void validateLogicalTableName(
+      LogicalTableConfig logicalTableConfig,
+      Predicate<String> physicalTableExistsPredicate,
+      Predicate<String> brokerTenantExistsPredicate,
+      ZkHelixPropertyStore<ZNRecord> propertyStore) {
```

Comment:
Can you make this also a `Predicate` to hide implementation details?

Reply:
We can do it, but one advantage I see is, if `propertyStore` can be reused in future for any validation. What do you think about that?

Reply:
Sounds good